drop-in/src/main/java/com/adyen/checkout/dropin/service/DropInServiceResult.kt

```diff
@@ -31,8 +38,46 @@ sealed class DropInServiceResult : BaseDropInServiceResult() {
     /**
      * Call was successful and returned with an
      * [com.adyen.checkout.components.model.payments.response.Action] that needs to be handled.
+     *
+     * Use [com.adyen.checkout.components.model.payments.response.Action.SERIALIZER] to serialize
+     * your JSON response string.
+     */
+    @Suppress("MemberNameEqualsClassName")
+    class Action : DropInServiceResult {
+        val action: com.adyen.checkout.components.model.payments.response.Action
+
+        constructor(action: com.adyen.checkout.components.model.payments.response.Action) {
+            this.action = action
+        }
+
+        @Deprecated("Use the new constructor which takes an Action object as parameter")
+        constructor(actionJSON: String) {
+            val actionJSONObject = try {
+                JSONObject(actionJSON)
+            } catch (e: JSONException) {
+                throw CheckoutException("Provided action is not a JSON object")
+            }
+            action = com.adyen.checkout.components.model.payments.response.Action.SERIALIZER.deserialize(actionJSONObject)
+        }
+    }
+
+    /**
+     * Only applicable for gift card flow.
+     *
+     * Update drop-in with a new list of payment methods and optionally an order.
+     *
+     * After submitting a partial payment, you need to call /paymentMethods again with the new remaining payment amount, and
+     * pass the updated payment methods list, alongside the latest order object.
+     *
+     * Also after cancelling an order, you need to call /paymentMethods again with the original payment amount, and pass the
+     * updated payment methods list, with a null order object.
+     *
+     * Use [OrderResponse.SERIALIZER] to serialize your JSON response string.
+     *
+     * @param paymentMethodsApiResponse the updated payment methods list.
+     * @param order the order object returned from the backend, or null if an order was cancelled.
      */
-    class Action(val actionJSON: String) : DropInServiceResult()
+    class Update(val paymentMethodsApiResponse: PaymentMethodsApiResponse, val order: OrderResponse?) : DropInServiceResult()
```

Comment:
i think we can enforce merchants to pass a non-null `OrderResponse` here. as far as i understood this is for partially paid orders and there has to be an order in the response to know we're at this step. thus we can make the `OrderResponse` non-null. i think by doing this we force them to not use `Update` when they don't have an order

Reply:
I made it nullable so that they can call this `Update` again after order cancellation (not implemented yet) and pass new payment methods without an order.